views/layouts/MainLayout.jsx

```diff
@@ -1,52 +1,85 @@
 import React, { useState, useEffect, useContext } from 'react'
-import { Outlet, useLocation, useNavigate } from 'react-router-dom'
+import { Outlet, useLocation, useNavigate, useParams } from 'react-router-dom'
 import classNames from 'classnames'
 import Header from '../components/Header'
 import Sidebar from '../components/Sidebar'
 
 import api from '../api'
 import getCounts from '../fe-utils/countUtil'
 import { fetchSubjects } from '../fe-utils/fetchUtil'
-import { AuthContext, NotificationContext } from '../contexts'
+import {
+  AuthContext,
+  ConfigurationsContext,
+  NotificationContext,
+  SidebarContext,
+} from '../contexts'
 import { headerTitle } from './helpers'
 
+const TEMPORARY_SIDEBAR = 'temporary'
+const PERSISTENT_SIDEBAR = 'persistent'
+const dashboard = 'dashboard'
+
 const MainLayout = ({ classes, theme }) => {
+  const [configurations, setConfigurations] = useContext(ConfigurationsContext)
   const [, setNotification] = useContext(NotificationContext)
+  const [openSidebar, setOpenSidebar] = useContext(SidebarContext)
   const [user, setUser] = useContext(AuthContext)
   const [users, setUsers] = useState([])
+  const [persistSidebar, setPersistSidebar] = useState(PERSISTENT_SIDEBAR)
```

Comment:
why is the name of this state `persistSidebar` when it could be either persistent or temporary? Should it be `drawerVariant` or something else?

Reply:
Thanks for this. 
[36fa637](https://github.com/AMP-SCZ/dpdash/pull/352/commits/36fa6377bc96de2b1ecbc84b556343889d45c2f3)